src/w2grid.js

```diff
@@ -2572,7 +2572,7 @@
                                 this.last.xhr_hasMore = true;
                             } else {
                                 this.last.xhr_hasMore = false;
-                                this.total = this.last.xhr_offset + data.records.length;
+                                this.total = this.offset + this.last.xhr_offset + data.records.length;
```

Comment:
Are you sure it is correct? This.last.xhr_offset is set before call is made.

Reply:
In the test case `this.last.xhr_offset` is less than `this.offset`. I don't know what `this.last.xhr_offset` is supposed to be. I posted further information in the [issue](https://github.com/vitmalina/w2ui/issues/1775#issuecomment-506269214).